plugin/web/extensions/infobox/core/components/viewer/index.tsx

```diff
@@ -0,0 +1,95 @@
+import type { Primitive, PublicSetting } from "@web/extensions/infobox/types";
+import { Collapse } from "@web/sharedComponents";
+import { styled } from "@web/theme";
+import { useEffect, useState } from "react";
+
+type Props = {
+  key: number;
+  primitive: Primitive;
+  publicSettings: PublicSetting[];
+};
+
+type PropertyItemType = {
+  key: string;
+  title: string;
+  value?: any;
+};
+
+const Viewer: React.FC<Props> = ({ primitive, publicSettings, key, ...props }) => {
+  const [propertyList, setPropertyList] = useState<PropertyItemType[]>([]);
+
+  useEffect(() => {
+    const propertyItem: PropertyItemType[] = [];
+    const publicSetting = publicSettings.find(ps => ps.type === primitive.type);
+    if (!publicSetting || publicSetting.properties.length === 0) {
+      primitive.properties.forEach(property => {
+        propertyItem.push({
+          ...property,
+          title: property.key,
+        });
+      });
+    } else {
+      publicSetting.properties
+        .filter(psp => !psp.hidden)
```

Comment:
Sorry I could be wrong, but won't this hide hidden (non-public) properties ALWAYS? Even if editor mode?